src/loader/cmi.cppo.ml

```diff
@@ -1037,6 +1037,7 @@ and read_signature_noenv env parent (items : Odoc_model.Compat.signature) =
             else shadowed
           in
             loop (ModuleType mtd :: acc, shadowed) rest
+#if OCAML_VERSION < (5,1,0)
```

Comment:
You could restrict the ifdef to the pattern rather than duplicate the code, since we only need to ignore ghost items.

Reply:
I've changed this to just ifdef the pattern.